extensions/lisp-mode/completion.lisp

```diff
@@ -7,6 +7,8 @@
            :region-completion))
 (in-package :lem-lisp-mode/completion)
 
+(defvar *documentation-popup-gravity* :vertically-adjacent-window-dynamic)
```

Comment:
Even if the setting might not be user facing, a docstring would help documentors and users: what does this solve and what are the possible values? 🙏

Reply:
I would actually like this setting to be user facing, so you are right some documentation is necessary.
Is there a better way to expose user facing variables?
Does lem have something like settings page that I missed?

Reply:
would you be fine with this as documentation?
```lisp
(setf (documentation '*documentation-popup-gravity* 'variable)
      (format nil "Set the gravity of the documentation popup. Possibilities are:~%~s"
              (handler-case (lem/popup-window::ensure-gravity nil)
                (condition (arg) (cdr (type-error-expected-type arg))))))
```
I don't really want people to have to change the documentation just because they added a new gravity, since it's not really related to the gravities themselves.

Reply:
oh, this, really? cxxxr will judge^^

> set the gravity

I didn't really follow what this does, so I'm a good guinea ping: what does the gravity solve for the documentation popup?

"Set the gravity of the documentation popup. By default, we ensure the popup is displayed *like this* to prevent *this*."

It might enough to give a couple default possibilities and refer to a function to know more.

Future doc generators might want an existing docstring in the `defvar`. It really depends.

---

You can pick cxxxr as reviewer when you're ready.

Reply:
For now, it is better to prepare docstrings as static strings.

Reply:
I understand. I was mostly intrigued by the possibility of programmatic doc-strings, it does seem like an extremely powerful feature to autogenerate part of the documentation.

I hope the new doc-string is more helpful. There should probably be some documentation for grativies in general. But I'm not sure how it can be presented well to the user, since gravities are created from keyword symbols. One could do something along the lines of: `(setf (documentation :my-cool-gravity 'variable) "the best gravity")` for all the gravity-keywords, lem would then display the doc-string during completion.

Slightly off topic: Are there any plans for a better config-system? Figuring out what can be set in `.lem/config.lisp` is somewhat annoying.